File: Sismique/Models/Image.cs
using MetadataExtractor;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ARIO.Models
{
    public class Image
    {
        public int ID { get; set; }

        [Required] public string File { get; set; }

        [Required] public string Thumbnail { get; set; }

        [Required] public DateTime Date { get; set; }

        [Required] public double Latitude { get; set; }

        [Required] public double Longitude { get; set; }

        public int ReportId { get; set; }

        public Report Report { get; set; }

        public ICollection<ImageCategory> ImageCategories { get; } = new List<ImageCategory>();

        public bool HasGpsCoordinates()
        {
            return Latitude != 0 && Longitude != 0;
        }

        public GeoLocation GetGeoLocation()
        {
            return new GeoLocation(Latitude, Longitude);
        }
    }
}

File: Sismique/ClientApp/app/components/shared/image.ts
import { GeoLocation } from "./geolocation";
import { IImageCategoryProperties, ImageCategory } from "./imagecategory";
import { IReportProperties, Report } from "./report";

export interface IImageProperties {
    id: number;
    file: string;
    thumbnail: string;
    date: Date;
    latitude: number;
    longitude: number;
    reportId: number;
    report: IReportProperties;
    imageCategories: IImageCategoryProperties[];
}

export class Image {
    id: number;
    file: string;
    thumbnail: string;
    date: Date;
    latitude: number;
    longitude: number;
    reportId: number;
    report: Report;
    imageCategories: ImageCategory[];

    deserialize(input: IImageProperties): Image {
        this.id = input.id;
        this.file = input.file;
        this.thumbnail = input.thumbnail;
        this.date = new Date(input.date);
        this.latitude = input.latitude;
        this.longitude = input.longitude;
        this.reportId = input.reportId;

        if (input.report != null) {
            this.report = new Report();
            this.report.deserialize(input.report);
        }

        this.imageCategories = [];
        if (input.imageCategories != null) {
            for (let ic of input.imageCategories) {
                const imageCategories = new ImageCategory();
                imageCategories.deserialize(ic);
                this.imageCategories.push(imageCategories);
            }
        }

        return this;
    }

    hasGpsCoordinates(): boolean {
        return this.latitude !== 0.0 && this.longitude !== 0.0;
    }

    getGeoLocation(): GeoLocation {
        return new GeoLocation(this.latitude, this.longitude);
    }
}
